mcs/class/System/Mono.AppleTls/SecureTransport.cs

```diff
@@ -0,0 +1,251 @@
+		// Ssl_3_0_only = 3,         
+		// Tls_1_0_only = 5,         
+		// All = 6,                
+	}
+
+	// subset of OSStatus (int)
+	enum SslStatus {
+		Success					= 0,		// errSecSuccess in SecBase.h
+		Protocol				= -9800,
+		Negotiation				= -9801,
+		FatalAlert				= -9802,
+		WouldBlock				= -9803,
+		SessionNotFound				= -9804,
+		ClosedGraceful				= -9805,
+		ClosedAbort				= -9806,
+		XCertChainInvalid			= -9807,
+		BadCert					= -9808,
+		Crypto					= -9809,
+		Internal				= -9810,
+		ModuleAttach				= -9811,
+		UnknownRootCert				= -9812,
+		NoRootCert				= -9813,
+		CertExpired				= -9814,
+		CertNotYetValid				= -9815,
+		ClosedNotNotified			= -9816,
+		BufferOverflow				= -9817,
+		BadCipherSuite				= -9818,
+		PeerUnexpectedMsg			= -9819,
+		PeerBadRecordMac			= -9820,
+		PeerDecryptionFail			= -9821,
+		PeerRecordOverflow			= -9822,
+		PeerDecompressFail			= -9823,
+		PeerHandshakeFail			= -9824,
+		PeerBadCert				= -9825,
+		PeerUnsupportedCert			= -9826,
+		PeerCertRevoked				= -9827,
+		PeerCertExpired				= -9828,
+		PeerCertUnknown				= -9829,
+		IllegalParam				= -9830,
+		PeerUnknownCA				= -9831,
+		PeerAccessDenied			= -9832,
+		PeerDecodeError				= -9833,
+		PeerDecryptError			= -9834,
+		PeerExportRestriction			= -9835,
+		PeerProtocolVersion			= -9836,
+		PeerInsufficientSecurity		= -9837,
+		PeerInternalError			= -9838,
+		PeerUserCancelled			= -9839,
+		PeerNoRenegotiation			= -9840,
+		PeerAuthCompleted			= -9841, // non fatal
+		PeerClientCertRequested			= -9842, // non fatal
+		HostNameMismatch			= -9843,
+		ConnectionRefused			= -9844,
+		DecryptionFail				= -9845,
+		BadRecordMac				= -9846,
+		RecordOverflow				= -9847,
+		BadConfiguration			= -9848,
+		UnexpectedRecord			= -9849,
+		SSLWeakPeerEphemeralDHKey               = -9850,
+		SSLClientHelloReceived                  = -9851 // non falta
+	}
+
+	// Security.framework/Headers/SecureTransport.h
+	// untyped enum
+	enum SslSessionOption {
+		BreakOnServerAuth,
+		BreakOnCertRequested,
+		BreakOnClientAuth,
+
+		// FalseStart,
+
+		// SendOneByteRecord,
+
+		// AllowServerIdentityChange = 5,
+		
+		// Fallback = 6,
+
+		// BreakOnClientHello = 7,
+
+		// AllowRenegotiation = 8,
+	}
+
+	// Security.framework/Headers/SecureTransport.h
+	// untyped enum
+	enum SslAuthenticate {
+		// Never,
+		// Always,
+		Try = 2,
+	}
+
+	// Security.framework/Headers/SecureTransport.h
+	// untyped enum
+	enum SslProtocolSide {
+		Server,
+		Client,
+	}
+
+	// Security.framework/Headers/SecureTransport.h
+	// untyped enum
+	enum SslConnectionType {
+		Stream,
+		// Datagram
+	}
+
+	// Security.framework/Headers/SecureTransport.h
+	// untyped enum
+	enum SslSessionState {
+		Invalid = -1,
+		// Idle,
+		// Handshake,
+		// Connected,
+		// Closed,
+		// Aborted
+	}
+
+	// Security.framework/Headers/SecureTransport.h
+	// untyped enum
+	enum SslClientCertificateState {
+		None,
+		Requested,
+		Sent,
+		Rejected
+	}
+
+	// Security.framework/Headers/CipherSuite.h
```

Comment:
Cosmetic post-landing:
Replace this enum with `Mono.Security.Interface.CipherSuiteCode`.